enhancements/microshift/microshift-csi-snapshot-integration.md

```diff
@@ -0,0 +1,455 @@
+guaranteed backwards compatible within each major version.  Therefore, micro-version upgrades will not require user 
+intervention to protect snapshots from deletion or orphaning.
+
+Upgrading the CSI containers will introduce a small interrupt window to provisioning operations.  Therefore, 
+to mitigate any risk or orphaning resources, it is recommended that snapshotting be allowed to complete before 
+restarting MicroShift and initiating the upgrade.
+
+#### Configuring
+
+Cluster admins will use the VolumeSnapshotClass API to set dynamic configurations for snapshot creation. 
+Driver-specific configuration is available through the ‘parameters’ sub-field, which is a string:string map and is 
+defined by the particular storage provider.
+
+#### Deploying Applications
+
+### API Extensions
+
+The APIs to be added are defined under the `snapshot.storage.k8s.io/v1` API group. They are:
+
+1. VolumeSnapshot
+2. VolumeSnapshotClass
+3. VolumeSnapshotContent
+
+For a detailed description of these APIs, see 
+[OpenShift documentation](https://docs.openshift.com/container-platform/4.13/storage/container_storage_interface/persistent-storage-csi-snapshots.html#volume-snapshot-crds)
+
+### Risks and Mitigations
+
+- Increased overhead.  The CSI snapshot controller and sidecar increase the on-disk footprint by a
+total of 360MB.  At idle, the components use a negligible amount of CPU (>1% CPU on a 2 core machine)
+and roughly 18Mb.
+
+### Drawbacks
+
+## Design Details
+
+### CSI Snapshot Controller
+
+The CSI Snapshot Controller is maintained SIG-Storage upstream. Downstream OCP releases of this container 
+image will be used by MicroShift.  The controller is a standalone component responsible for watching 
+`volumeSnapshot`, `volumeSnapshotContent`, and `volumeSnapshotClass` objects.  When a `volumeSnapshot` is created, 
+the controller will trigger a snapshot operation by creating a `volumeSnapshotConent` object, which will be 
+detected by the _CSI external-snapshotter sidecar_.
+
+### CSI External Snapshotter
+
+The CSI External Snapshotter is maintained SIG-Storage upstream. Downstream OCP releases of this container 
+image will be used by MicroShift. The container is deployed as a sidecar of the _topolvm-controller_. It 
+watches for `volumeSnapshotContent` and `volumeSnapshotClass` events.  When an event is detected, it 
+makes the appropriate CSI gRPC call to LVMS via a shared unix socket.
+
+### CSI Validation Webhook
+
+The CSI Snapshot Validation Webhook is maintained SIG-Storage upstream. Downstream OCP releases of this container 
+image will be used by MicroShift.  The webhook serves as gatekeeper to `volumeSnapshot` CREATE and UPDATE events. 
+For the specifics of validation, _see_ 
+[Kubernetes KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1900-volume-snapshot-validation-webhook#validating-scenarios).
```

Comment:
There's not reference to the validation webhook in OpenShift docs, and it's only briefly mentioned in K8s docs.  The enhancement is the best source for details about it.